### PR TITLE
[UTY-2393] Port build system module to new CodeWriter 

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

Port build system module to new CodeWriter

#### Tests

check generated code diff

#### Documentation

- [x] changelog

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: test-project/Assets/Generated/Source/improbable/testschema/ExhaustiveSingularComponentReaderWriter.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
-// ===========
+// =====================================================
```
Comment (jamiebrynes7) [inline test-project/Assets/Generated/Source/improbable/testschema/ExhaustiveSingularComponentReaderWriter.cs]: Why are all of these changed in the build system module update..?

File: workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/UnityWorkerMenuGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+                        buildWorkerMenu.Method("private static void MenuBuildLocal(string[] filteredWorkerTypes)", m =>
+                        {
+                            m.Line("WorkerBuilder.MenuBuild(BuildEnvironment.Local, filteredWorkerTypes);");
+                        });
+
+                        buildWorkerMenu.Method("private static void MenuBuildCloud(string[] filteredWorkerTypes)", m =>
+                        {
+                            m.Line("WorkerBuilder.MenuBuild(BuildEnvironment.Cloud, filteredWorkerTypes);");
+                        });
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/UnityWorkerMenuGenerator.cs]: Reading through this again - these seem mostly redundant

Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/UnityWorkerMenuGenerator.cs]: yeah - could just call `WorkerBuild.MenuBuild` directly

File: test-project/Assets/Generated/Source/improbable/testschema/ExhaustiveSingularComponentReaderWriter.cs (hunk 1 above)
Comment (paulbalaji) [reply test-project/Assets/Generated/Source/improbable/testschema/ExhaustiveSingularComponentReaderWriter.cs]: forgot to generate test-project before the final codewriter pr (where the header changed)

this PR is also not based off the test-project port PR  - hence the ===== in this and #1241

File: workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/UnityWorkerMenuGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+                        buildWorkerMenu.Method("private static void MenuCleanAll()", m =>
```
Comment (zeroZshadow) [inline workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/UnityWorkerMenuGenerator.cs]: Would be nice to have an overload for these to straight up pass in an `IEnumerator<string>` or a delegate that returns one.

This lets you write the following:

```
buildWorkerMenu.Method("private static void MenuCleanAll()", new[]
{
    "WorkerBuilder.Clean();",
    "Debug.Log(\"Clean completed\");"
});

buildWorkerMenu.Method("private static void MenuCleanAll()", () => new[]
{
    "WorkerBuilder.Clean();",
    "Debug.Log(\"Clean completed\");"
})
```

File: workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/UnityWorkerMenuGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                        });
+
+                        for (var i = 0; i < workerTypes.Count; i++)
+                        {
+                            var workerType = workerTypes[i];
+                            var workerTypeString = $@"""{workerType}""";
+
+                            buildWorkerMenu.Method($"public static void BuildLocal{workerType}()", m =>
+                            {
+                                m.Annotate($@"MenuItem(EditorConfig.ParentMenu + ""/"" + LocalMenu + ""/{workerType}"", false, EditorConfig.MenuOffset + {i})");
```
Comment (zeroZshadow) [inline workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/Generators/BuildSystem/UnityWorkerMenuGenerator.cs]: As discussed, maybe change the api to allow this:

```
buildWorkerMenu.Annotate($@"MenuItem(EditorConfig.ParentMenu + ""/Clean all workers"", false, EditorConfig.MenuOffset + {workerTypes.Count})")
    .Method("public static void Clean()", new[]
    {
        "MenuCleanAll();"
    });
```

**Pull request merged**